dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -142,23 +151,34 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
 
     winrt::Windows::Foundation::IInspectable PushNotificationManager::Deserialize(winrt::Windows::Foundation::Uri const& uri)
     {
-        // Verify if the uri contains a background notification payload.
-        // Otherwise, we expect to process the notification in a background task.
-        for (auto const& pair : uri.QueryParsed())
+        winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs eventArgs{ nullptr };
+
+        // All packaged processes are triggered through COM via Long Running Process or the Background Infra OS component
+        if (AppModel::Identity::IsPackagedProcess())
+        {
+            THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_TIMEOUT), !m_waitHandleForArgs.wait(c_receiveArgsTimeoutInMSec));
+            auto lock{ m_lock.lock_shared() };
+            THROW_HR_IF(E_UNEXPECTED, !m_backgroundTaskArgs);
+            eventArgs = m_backgroundTaskArgs;
+        }
+        else // The process was launched via ShellExecute and we need to parse the uri (Only unpackaged)
         {
-            if (pair.Name() == L"payload")
+            for (auto const& pair : uri.QueryParsed())
             {
-                // Convert escaped components to its normal content
-                // from the conversion done in the LRP (see NotificationListener.cpp)
-                std::wstring payloadAsEscapedWstring{ pair.Value() };
-                std::wstring payloadAsWstring{ winrt::Windows::Foundation::Uri::UnescapeComponent(payloadAsEscapedWstring) };
-                return winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(payloadAsWstring);
+                if (pair.Name() == L"payload")
+                {
+                    // Convert escaped components to its normal content
+                    // from the conversion done in the LRP (see NotificationListener.cpp)
+                    std::wstring payloadAsEscapedWstring{ pair.Value() };
```

Comment:
Shouldn't need this temporary.  `pair.Value()` is of type `winrt::hstring` and `UnescapeComponent` takes hstring.  When possible, avoid copying between string types - and avoid copying strings.

Consider:

```c++
auto unescapedString = winrt::Windows::Foundation::Uri::UnescapeComponent(pair.Value());
```

... and then make sure PushNotificationReceivedEventArgs's ctor accepts `winrt::hstring`

Reply:
Done. I updated all the methods to use hstring. There is one API helper at the bottom of the stack which still requires a change to wstring but changing that would impact code elsewhere so I decided to leave as is.